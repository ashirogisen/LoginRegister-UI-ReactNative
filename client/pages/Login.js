import React, { useState } from 'react'
import { 
  StyleSheet, 
  View, 
  Text, 
  TouchableOpacity 
} from 'react-native'

import Logo from '../components/Logo'
import Form from '../components/Form'
import Button from '../components/Button'

function Login({ navigation }) {
  const [ email, setEmail ] = useState('')
  const [ password, setPassword ] = useState('')

  const toCreateAccount = () => {
    navigation.navigate('CreateAccount')
  }

  const handleSubmit = () => {
    console.log(email)
    console.log(password)
  }

  const styles = StyleSheet.create({
    mainContainer: {
      backgroundColor: '#555555',
      height: '100%',
      justifyContent: 'center',
      alignItems: 'center'
    },
    formsContainer: {
      width: '90%',
      justifyContent: 'center',
      alignItems: 'center',
      padding: 0,
      marginVertical: 45
    },
    buttonContainer: {
      width: '90%'
    },
    footerContainer: {
      padding: 30,
      marginTop: 14,
      width: '100%',
      flexDirection: 'row',
      justifyContent: 'space-between',
      alignItems: 'center'
    },
    footerMenus: {
      color: 'white'
    }
  })

  const buttonCustomStyles = {
    backgroundColor: '#888888',
    justifyContent: 'center',
    alignItems: 'center'
  }

  const forms = [
    { 
      value: email, 
      setValue: setEmail, 
      icon: 'md-mail', 
      placeholder: 'Email', 
      iconColor: 'white' 
    },
    { value: password, 
      setValue: setPassword, 
      icon: 'md-lock', 
      placeholder: 'Password', 
      iconColor: 'white' 
    }
  ]

  return(
    <View testID="test-mainContainer" style={styles.mainContainer}>
      <Logo />
      <View testID="test-formsContainer" style={styles.formsContainer}>
        {
          forms.map((form, index) => (
            <Form key={index} {...form} />
          ))
        }
      </View>
      <View testID="test-buttonContainer" style={styles.buttonContainer}>
        <Button 
          onSubmit={handleSubmit} 
          customStyles={buttonCustomStyles} 
          name="GET STARTED" 
        />
      </View>
      <View testID="test-footerContainer" style={styles.footerContainer}>
        <TouchableOpacity onPress={toCreateAccount}>
          <Text testID="test-footerMenu-1" style={styles.footerMenus}>Create Account</Text>
        </TouchableOpacity>
        <TouchableOpacity>
          <Text testID="test-footerMenu-2" style={styles.footerMenus}>Forgot Password</Text>
        </TouchableOpacity>
      </View>
    </View>
  )
}

export default Login