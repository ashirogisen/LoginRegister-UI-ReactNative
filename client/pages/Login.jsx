import React, { useState } from 'react'
import { StyleSheet, View, Text, TouchableOpacity } from 'react-native'

import Logo from '../components/Logo'
import Form from '../components/Form'
import Button from '../components/Button'

function Login(props) {
  const [ email, setEmail ] = useState('')
  const [ password, setPassword ] = useState('')

  const toCreateAccount = () => {
    props.navigation.navigate('CreateAccount')
  }

  const TestFunction = () => {
    console.log(email)
    console.log(password)
  }

  const styles = StyleSheet.create({
    mainContainer: {
      backgroundColor: '#555555',
      height: '100%',
      justifyContent: 'center',
      alignItems: 'center'
    },
    formsContainer: {
      width: '90%',
      justifyContent: 'center',
      alignItems: 'center',
      padding: 0,
      marginVertical: 45
    },
    buttonContainer: {
      width: '90%'
    },
    footerContainer: {
      padding: 30,
      marginTop: 14,
      width: '100%',
      flexDirection: 'row',
      justifyContent: 'space-between',
      alignItems: 'center',
    },
    footerMenus: {
      color: 'white'
    }
  })

  const buttonCustomStyles = {
    backgroundColor: '#888888',
    justifyContent: 'center',
    alignItems: 'center',
  }

  return(
    <View style={styles.mainContainer}>
      <Logo />
      <View style={styles.formsContainer}>
        <Form value={email} setValue={setEmail} icon="md-mail" placeholder="Email"/>
        <Form value={password} setValue={setPassword} icon="md-lock" placeholder="Password"/>
      </View>
      <View style={styles.buttonContainer}>
        <Button command={TestFunction} customStyles={buttonCustomStyles} name="GET STARTED" />
      </View>
      <View style={styles.footerContainer}>
        <TouchableOpacity onPress={toCreateAccount}>
          <Text style={styles.footerMenus}>Create Account</Text>
        </TouchableOpacity>
        <TouchableOpacity>
          <Text style={styles.footerMenus}>Forgot Password</Text>
        </TouchableOpacity>
      </View>
    </View>
  )
}

export default Login