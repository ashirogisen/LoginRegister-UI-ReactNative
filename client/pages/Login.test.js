import React from 'react'
import Login from './Login'
import renderer from 'react-test-renderer'
import { render } from 'react-native-testing-library'

describe('Render Login', () => {
  it('Renders without crashing', () => {
    const tree = renderer.create(
      <Login />
    ).toJSON();
    expect(tree).toMatchSnapshot()
  })

  it('Should render main container, button container, forms container, footers container, and footers menus', () => {
    const { getByTestId } = render (
      <Login />
    )
    expect(getByTestId('test-buttonContainer')).toBeTruthy()
    expect(getByTestId('test-formsContainer')).toBeTruthy()
    expect(getByTestId('test-mainContainer')).toBeTruthy()
    expect(getByTestId('test-footerMenu-2')).toBeTruthy()
    expect(getByTestId('test-footerMenu-1')).toBeTruthy()
  })
})
