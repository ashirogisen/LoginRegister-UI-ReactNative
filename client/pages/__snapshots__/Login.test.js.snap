// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Login Renders without crashing 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#555555",
      "height": "100%",
      "justifyContent": "center",
    }
  }
  testID="test-mainContainer"
>
  <View
    testID="test-logoContainer"
  >
    <Image
      source={1}
      style={
        Object {
          "height": 100,
          "width": 330,
        }
      }
      testID="test-logoImage"
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginVertical": 45,
        "padding": 0,
        "width": "90%",
      }
    }
    testID="test-formsContainer"
  >
    <View
      accessible={true}
      focusable={false}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#888888",
          "borderBottomWidth": 1,
          "borderColor": "#D9D5DC",
          "borderLeftWidth": 1,
          "borderRadius": 30,
          "borderRightWidth": 1,
          "borderTopWidth": 1,
          "borderWidth": 1,
          "color": "white",
          "flexDirection": "row",
          "margin": 10,
          "marginLeft": 2,
          "opacity": 1,
          "width": "100%",
        }
      }
      testID="test-formItem"
    >
      <Text />
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        editable={true}
        keyboardType="default"
        onChangeText={[Function]}
        placeholder="Email"
        placeholderTextColor="#bdc3c7"
        rejectResponderTermination={true}
        style={
          Object {
            "color": "#000",
            "flex": 1,
            "fontSize": 17,
            "height": 50,
            "paddingLeft": 8,
            "paddingRight": 5,
            "top": 1.5,
          }
        }
        testID="test-formInput"
        underlineColorAndroid="rgba(0,0,0,0)"
        value=""
      />
    </View>
    <View
      accessible={true}
      focusable={false}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#888888",
          "borderBottomWidth": 1,
          "borderColor": "#D9D5DC",
          "borderLeftWidth": 1,
          "borderRadius": 30,
          "borderRightWidth": 1,
          "borderTopWidth": 1,
          "borderWidth": 1,
          "color": "white",
          "flexDirection": "row",
          "margin": 10,
          "marginLeft": 2,
          "opacity": 1,
          "width": "100%",
        }
      }
      testID="test-formItem"
    >
      <Text />
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        editable={true}
        keyboardType="default"
        onChangeText={[Function]}
        placeholder="Password"
        placeholderTextColor="#bdc3c7"
        rejectResponderTermination={true}
        style={
          Object {
            "color": "#000",
            "flex": 1,
            "fontSize": 17,
            "height": 50,
            "paddingLeft": 8,
            "paddingRight": 5,
            "top": 1.5,
          }
        }
        testID="test-formInput"
        underlineColorAndroid="rgba(0,0,0,0)"
        value=""
      />
    </View>
  </View>
  <View
    style={
      Object {
        "width": "90%",
      }
    }
    testID="test-buttonContainer"
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#888888",
          "borderBottomWidth": null,
          "borderColor": "#007aff",
          "borderLeftWidth": null,
          "borderRadius": 57,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "borderWidth": undefined,
          "elevation": 2,
          "flexDirection": "row",
          "height": 45,
          "justifyContent": "center",
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
          "shadowColor": undefined,
          "shadowOffset": undefined,
          "shadowOpacity": undefined,
          "shadowRadius": undefined,
          "width": "100%",
        }
      }
      testID="test-button"
    >
      <Text
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#fff",
            "fontFamily": "System",
            "fontSize": 16.5,
            "marginLeft": 0,
            "marginRight": 0,
            "paddingLeft": 16,
            "paddingRight": 16,
          }
        }
        testID="test-buttonText"
        uppercase={false}
      >
        GET STARTED
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginTop": 14,
        "padding": 30,
        "width": "100%",
      }
    }
    testID="test-footerContainer"
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
          }
        }
        testID="test-footerMenu-1"
      >
        Create Account
      </Text>
    </View>
    <View
      accessible={true}
      focusable={false}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
          }
        }
        testID="test-footerMenu-2"
      >
        Forgot Password
      </Text>
    </View>
  </View>
</View>
`;
